classes/controller/menu/class-archive-import.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * An importer for archived content from an XML file.
+ *
+ * @package P4GBKS\Controllers\Menu
+ */
+
+namespace P4GBKS\Controllers\Menu;
+
+use WP_Query;
+
+/**
+ * An importer for archived content from an XML file.
+ *
+ * @package P4GBKS\Controllers\Menu
+ */
+class Archive_Import extends Controller {
+
+	private const WAY_BACK_URL = 'https://wayback.archive-it.org/9650/';
+
+	private const P3_BASE_URL = 'http://p3-raw.greenpeace.org/';
+
+	/**
+	 * Get all urls of P3 that were already imported.
+	 *
+	 * @return string[] All urls of P3 that were already imported.
+	 */
+	private static function get_imported_p3_urls(): array {
+		$query = new WP_Query(
+			[
+				'meta_key'       => 'p3_url',
+				'posts_per_page' => -1,
+			]
+		);
+
+		$urls = [];
+		foreach ( $query->posts as $post ) {
+			$urls[] = get_post_meta( $post->ID, 'p3_url' )[0];
+		}
+
+		return $urls;
+	}
+
+	/**
+	 * Add the admin menu item.
+	 *
+	 * @return void
+	 */
+	public function create_admin_menu() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+		add_submenu_page(
+			'edit.php?post_type=archive',
+			__( 'Archive Import', 'planet4-blocks-backend' ),
+			__( 'Archive Import', 'planet4-blocks-backend' ),
+			'edit_posts',
+			'archive-import',
+			[ $this, 'get_page' ]
+		);
+	}
+
+	/**
+	 * Render the admin page.
+	 *
+	 * @return string|void The html for the admin page.
+	 */
+	public function get_page() {
+		// There is no button for this currently, needs to be manually added to the url.
+		if ( isset( $_GET['delete_all'] ) ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$archives = get_posts(
+				[
+					'post_type'   => 'archive',
+					'numberposts' => -1,
+				]
+			);
+			foreach ( $archives as $archive ) {
+				wp_delete_post( $archive->ID );
+			}
+			return 'Deleted all archived posts';
+		}
+
+		if ( isset( $_GET['done'] ) ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
I am not sure, when this condition executes? is it a part of XML import?

Reply:
Ah you're right, this isn't used anymore, will remove it.